### PR TITLE
Majority classifier modified such that it returns the same random cla…

### PR DESCRIPTION
**Pull request by mstrazar**
…ss for datasets with uniform class distributions. This is based on SHA1 hash function and should be consistent across systems.

### CODE REVIEW

File: Orange/classification/majority.py [hunk 1]
```diff
@@ -56,7 +71,11 @@ def predict(self, X):
         :return: predicted value
         :rtype: vector of majority values
         """
-        return tile(self.dist, (len(X), 1))
+        probs = tile(self.dist, (len(X), 1))
+        if self.unif_maj:
```
Comment (lanzagar) [inline Orange/classification/majority.py]: The if condition fails when self.unif_maj == 0.
Should be:

    if self.unif_maj is not None:

File: Orange/tests/test_majority.py [hunk 2]
```diff
@@ -55,10 +55,10 @@ def test_missing(self):
         iris = Table('iris')
         learn = MajorityLearner()
         for e in iris:
-            e.set_class("?")
+             e.set_class("?")
```
Comment (lanzagar) [inline Orange/tests/test_majority.py]: Was this intentional?

Comment (mstrazar) [inline Orange/tests/test_majority.py]: No, fixed.

File: Orange/classification/majority.py (hunk 1 above)
Comment (mstrazar) [inline Orange/classification/majority.py]: Fixed.

**Comment by astaric** [comment]
Manually merged in bf41cfc.

**Pull request closed without merging**